test/e2e/istio/istio_e2e_suite_test.go

```diff
@@ -0,0 +1,261 @@
+})
+
+// initialize all tests
+var _ = allTests()
+
+func allTests() bool {
+	/*
+		test order matters here:
+		- we run extender tests first because coretests modify settings (extender)
+		- we run rbac tests last as rbactests modify rbac
+	*/
+	return Describe("enterprise helm chart", func() {
+		extendertests.InitializeTests()
+		coretests.InitializeTests()
+		rbactests.InitializeTests()
+	})
+}
+
+func deployAndRegisterEnterprise() {
+	// override Docker Host Addr for CircleCI
+	extensions.DockerHostAddress = pkg.DockerHostAddress
+
+	err := installEnterpriseChart()
+	Expect(err).NotTo(HaveOccurred())
+	err = registerCluster("mgmt-cluster")
+	Expect(err).NotTo(HaveOccurred())
+	err = registerCluster("remote-cluster")
+	Expect(err).NotTo(HaveOccurred())
+
+	err = applyCustomBootstrapPatch("remote-cluster", "bookinfo", "reviews-v3")
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func installEnterpriseChart() error {
+	chartPath := os.Getenv("OUTPUT_CHART_PATH")
+	if chartPath == "" {
+		if err := runCommand("make", "package-chart"); err != nil {
+			return err
+		}
+		path, err := runCommandOut("make", "print-chart-path")
+		if err != nil {
+			return err
+		}
+		chartPath = path
+	}
+	licenseKey := os.Getenv("LICENSE_KEY")
+	if chartPath == "" {
+		return eris.Errorf("must provide var LICENSE_KEY with valid enterprise license key")
+	}
+
+	if err := runCommand(
+		"kubectl",
+		"create",
+		"ns",
+		"gloo-mesh",
+	); err != nil && !strings.Contains(err.Error(), `namespaces "gloo-mesh" already exists`) {
+		return err
+	}
+
+	if err := runCommand(
+		"helm",
+		"upgrade",
+		"--install",
+		"--namespace=gloo-mesh",
+		"gloo-mesh-enterprise",
+		chartPath,
+		"--set",
+		"licenseKey="+licenseKey,
+		"--set",
+		"gloo-mesh-ui.enabled=false", // disable apiserver/UI to free up compute for CI
+	); err != nil {
+		return err
+	}
+
+	for _, deployment := range []string{
+		"discovery",
+		"networking",
+		"enterprise-extender",
+		"rbac-webhook",
+	} {
+		if err := runCommand(
+			"kubectl",
+			"rollout",
+			"status",
+			"-n=gloo-mesh",
+			"deployment",
+			deployment,
+		); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// uses meshctl from path to register cluster;
+// make sure meshctl version matches gloo-mesh version from chart
+func registerCluster(cluster string) error {
+	apiServerAddr, err := getApiserverAddress(cluster)
+	if err != nil {
+		return err
+	}
+	if err := runCommand(
+		"meshctl",
+		"cluster",
+		"register",
+		"--mgmt-context=kind-mgmt-cluster",
+		"--cluster-name="+cluster,
+		"--remote-context=kind-"+cluster,
+		"--api-server-address="+apiServerAddr,
+	); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getApiserverAddress(cluster string) (string, error) {
+	switch runtime.GOOS {
+	case "darwin":
+		return "host.docker.internal", nil
+	default:
+		addr, err := runCommandOut("bash", "-c", `docker exec `+cluster+`-control-plane ip addr show dev eth0 | sed -nE 's|\s*inet\s+([0-9.]+).*|\1|p'`)
```

Comment:
can we reuse `DockerHostAddress` here?

Reply:
this isn't getting the address of the docker *host* but the docker *containers* for each kind cluster